modules/plink/vcf/main.nf

```diff
@@ -0,0 +1,44 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process PLINK_VCF {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::plink=1.90b6.21" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/plink:1.90b6.21--h779adbc_1"
+    } else {
+        container "quay.io/biocontainers/plink:1.90b6.21--h779adbc_1"
+    }
+
+    input:
+    tuple val(meta), path(vcf)
+
+    output:
+    tuple val(meta), path("*.bed"), emit: bed, optional: true
+    tuple val(meta), path("*.bim"), emit: bim, optional: true
+    tuple val(meta), path("*.fam"), emit: fam, optional: true
+
+    path "*.version.txt", emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+
+    """
+    plink \\
+        --vcf ${vcf} \\
```

Comment:
Just to confirm - there are no other optional input files that it can accept, correct?

Reply:
Doesn't look like it for vcf input. 
https://www.cog-genomics.org/plink/1.9/input#vcf

Reply:
Yeah, you are right, `plink` does accept quite a lot of input files. For example,

```
--bed <filename>
--bim <filename>
--fam <filename>
--ped <filename>
--map <filename>
--tped <filename>
--tfam <filename>
...
...
```

Given the experience gained from https://github.com/nf-core/modules/pull/669 , I am thinking that perhaps the best way ahead for the `plink` module would be to rely upon the a single `plink` module (without sub-tool option) to accomodate all these files as optional parameter. 

What are you thoughts?

Reply:
I think for usage it would be nicer to have modules for the different functionalities, e.g. `plink --vcf` and `plink --bed` since there are lots of options/inputs that are only compatible with certain flags, right? It would help decomplicate things rather than needing the programmer to figure out which inputs in the huge list need to be `[]` and which need to be provided. It would help with readability in my opinion.

Reply:
I do tend to agree on this one, having a process call like the following is really silly

```
PLINK ( xyz, [] , [] , [] , [] ...) 

```

Given that we do agree on the structure, are there any changes which need to be made to this PR?